PlayerScripts/PlayerScriptsPlace/StarterPlayer/StarterCharacterScripts/Sound.rbxmx

```diff
@@ -0,0 +1,381 @@
+	sound.Looped = looped	
+	
+	return sound
+end
+
+CreateNewSound("GettingUp", 	"rbxasset://sounds/action_get_up.mp3",					false, 	1)
+CreateNewSound("Died", 		"rbxasset://sounds/uuhhh.mp3",								false, 	1) 
+CreateNewSound("FreeFalling", 	"rbxasset://sounds/action_falling.mp3",					true,	1)
+CreateNewSound("Jumping", 		"rbxasset://sounds/action_jump.mp3",					false,	1)
+CreateNewSound("Landing", 		"rbxasset://sounds/action_jump_land.mp3",				false,	1)
+CreateNewSound("Splash", 		"rbxasset://sounds/impact_water.mp3",					false, 	1)
+CreateNewSound("Running", 		"rbxasset://sounds/action_footsteps_plastic.mp3",		true, 	1.85)
+CreateNewSound("Swimming", 	"rbxasset://sounds/action_swim.mp3",						true,	1.6)
+CreateNewSound("Climbing", 	"rbxasset://sounds/action_footsteps_plastic.mp3",			true,	1)]]></ProtectedString>
+		</Properties>
+		<Item class="LocalScript" referent="RBX35b0a371f02e4ae2baca5961c8b67265">
+			<Properties>
+				<bool name="Disabled">false</bool>
+				<Content name="LinkedSource"><null></null></Content>
+				<string name="Name">LocalSound</string>
+				<string name="ScriptGuid"></string>
+				<ProtectedString name="Source"><![CDATA[--[[
+	Author: @spotco
+	This script runs locally for the player of the given humanoid.
+	This script triggers humanoid sound play/pause actions locally.
+	
+	The Playing/TimePosition properties of Sound objects bypass FilteringEnabled, so this triggers the sound
+	immediately for the player and is replicated to all other players.
+	
+	This script has been optimized to reduce network traffic (in comparison to the existing humanoid sound scripts.
+]]--
+
+--All sounds are referenced by this ID
+local SFX = {
+	Died = 0;
+	Running = 1;
+	Swimming = 2;
+	Climbing = 3,
+	Jumping = 4;
+	GettingUp = 5;
+	FreeFalling = 6;
+	FallingDown = 7;
+	Landing = 8;
+	Splash = 9;
+}
+
+local Humanoid = nil
+local Head = nil
+
+--SFX ID to Sound object
+local Sounds = {}
+
+do
+	local Figure = script.Parent.Parent
+	Head = Figure:WaitForChild("Head")
+	Humanoid = Figure:WaitForChild("Humanoid")
+	
+	Sounds[SFX.Died] = 			Head:WaitForChild("Died")
+	Sounds[SFX.Running] = 		Head:WaitForChild("Running")
+	Sounds[SFX.Swimming] = 		Head:WaitForChild("Swimming")
+	Sounds[SFX.Climbing] = 		Head:WaitForChild("Climbing")
+	Sounds[SFX.Jumping] = 		Head:WaitForChild("Jumping")
+	Sounds[SFX.GettingUp] = 	Head:WaitForChild("GettingUp")
+	Sounds[SFX.FreeFalling] = 	Head:WaitForChild("FreeFalling")
+	Sounds[SFX.Landing] = 		Head:WaitForChild("Landing")
+	Sounds[SFX.Splash] = 		Head:WaitForChild("Splash")
+end
+
+local Util
+Util = {
+	
+	--Define linear relationship between (pt1x,pt2x) and (pt2x,pt2y). Evaluate this like at x.
+	YForLineGivenXAndTwoPts = function(x,pt1x,pt1y,pt2x,pt2y)
+		--(y - y1)/(x - x1) = m
+		local m = (pt1y - pt2y) / (pt1x - pt2x)
+		--float b = pt1.y - m * pt1.x;
+		local b = (pt1y - m * pt1x)
+		return m * x + b
+	end;
+	
+	--Clamps the value of "val" between the "min" and "max"
+	Clamp = function(val,min,max)
```

Comment:
re: I'll do another PR sometime with some more math_whatever functions.
I see lua_pushnumber and immediately go nope.